controllers/gatekeeper_controller.go

```diff
@@ -1472,59 +1488,61 @@ func setContainerAttrWithFn(obj *unstructured.Unstructured, containerFn func(map
 	return nil
 }
 
-func setContainerArg(
-	obj *unstructured.Unstructured, argName string, argValue string, isMultiArg bool,
+// updateContainerArg takes an object, container flag argument name and value to
+// set on the object, and a boolean whether the flag should be deleted, and
+// returns any resulting errors.
+func updateContainerArg(
+	obj *unstructured.Unstructured, argName string, argValue string, remove bool,
 ) error {
 	return setContainerAttrWithFn(obj, func(container map[string]interface{}) error {
 		args, found, err := unstructured.NestedStringSlice(container, "args")
 		if !found || err != nil {
 			return errors.Wrapf(err, "Unable to retrieve container arguments for: %s", managerContainer)
 		}
 
-		exists := false
+		// Determine whether this is a flag that can be declared more than once.
+		multiArgFlags := []string{
+			"--operation",
+			"--disable-opa-builtin",
+			"--exempt-namespace",
+			"--exempt-namespace-prefix",
+			"--exempt-namespace-suffix",
+			"--metrics-backend",
+		}
+		isMultiArg := slices.Contains(multiArgFlags, argName)
+
+		index := -1
 		for i, arg := range args {
 			n, v := util.FromArg(arg)
 			if n == argName && (!isMultiArg || isMultiArg && v == argValue) {
-				args[i] = util.ToArg(argName, argValue)
-				exists = true
+				index = i
 			}
 		}
 
-		if !exists {
+		if index != -1 && remove {
+			args = slices.Delete(args, index, index+1)
```

Comment:
How can we delete if multiple arguments should be delted? should be in range?

Reply:
The delete for multiple arguements requires a value to be specified.

Reply:
Right now there isn't a case I saw where more than one of a multiple argument flag should be deleted.

Reply:
Good!!